renku_notebooks/api/classes/server.py

```diff
@@ -68,13 +69,7 @@ def __init__(
         self.is_image_private = None
         self.image_workdir = None
         self.cloudstorage = cloudstorage
-        try:
-            self.gl_project = self._user.get_renku_project(
-                f"{self.namespace}/{self.project}"
-            )
-        except Exception as err:
-            current_app.logger.warning("Cannot find project because:", err)
-            self.gl_project = None
+        self.gl_project_name = f"{self.namespace}/{self.project}"
```

Comment:
Previously I fetched and defined the gitlab project here. And if the project did not exist (which is possible) this would fail. Now I check for the gitlab project only during creation when I call the property (which is cached).